app/support/pdf_font_helper.rb

```diff
@@ -0,0 +1,37 @@
+class PdfFontHelper
+
+  class << self
+
+    delegate :set_fonts, to: :new
+
+  end
```

Comment:
I think this construct is a bit weird for a helper—there is no state at all being manipulated in `set_fonts`, so why even create an object? Can we `def self.set_fonts` and eliminate this indirection in the implementation?

Reply:
Oh, right. The reason I did this is because making `set_fonts` a class method means the private methods need to be class methods as well. I'm open to doing something else, but this feels like a reasonable balance between usable API and readability.

Options: 
* Change the API to `PdfHelper.new.set_fonts(pdf)` or `PdfHelper.new(pdf).set_fonts`. Little more verbose, but not bad.
* Make all the methods (including the private methods) class methods. Little more verbose internally, and makes the internal helper methods public.

What do you think?

Reply:
What do you think about something like this?

```ruby
class PdfFontHelper

  FONT_DIR = Rails.root.join("app/assets/fonts")

  def self.set_fonts
    font_name = Settings.statement_pdf.font_name
    return if font_name.blank?

    pdf.font_families.update(
      font_name => {
        bold: FONT_DIR.join("#{font_name}-Bold.ttf"),
        italic: FONT_DIR.join("#{font_name}-Italic.ttf"),
        normal: FONT_DIR.join("#{font_name}-Regular.ttf"),
      }
    )
    pdf.font(font_name)
  end

end
```

I’m not sure there’s a need for maintaining `mappings` only to use it in `file_mappings` to create the new hash, so I’d get rid of the private methods entirely.

If you don’t like it and want to go with your original implementation, I’m okay with that too.

Reply:
I like that implementation. I had originally had the `mappings` method pulled out because I was thinking of allowing defining other variants (e.g. BoldItalic) through the settings, but then I decided that was overkill, but still left the method. I agree that with just the 3, it's fine to leave them inline and not doing the `reduce`.